src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1260,22 +1260,30 @@ private void EngineController_TraceReconcliationComplete(TraceReconciliationEven
             // This dictionary gets redistributed into a dictionary keyed by the workspace id.
 
             var workspaceOrphanMap = new Dictionary<Guid, List<string>>();
+            var nodeToWorkspaceMap = new Dictionary<Guid, WorkspaceModel>();
 
-            foreach (var ws in Workspaces.OfType<HomeWorkspaceModel>())
+            foreach (var maybeWs in Workspaces)
```

Comment:
why do this?

Reply:
I don't think there is any value in iterating all the nodes of a custom node workspace... maybe you can say a bit more about why you need this other map to improve performance.